EIPS/eip-zodiac.md

```diff
@@ -0,0 +1,175 @@
+1. **Avatars** are programmable Ethereum accounts, like the [Gnosis Safe](https://gnosis-safe.io). Avatars are the address that holds balances, owns systems, executes transaction, is referenced externally, and ultimately represents your DAO. Avatars MUST expose the `IAvatar` interface.
+2. **Modules** are contracts enabled by an Avatar that implement some control logic.
+3. **Modifiers** are contracts that sit between Modules and Avatars to modify the Module's behavior. For example, they might enforce a delay on all functions a Module attempts to execute or limit the scope of transactions that can be initiated by the module. Modifiers MUST expose the `IAvatar` interface.
+4. **Guards** are contracts that can be enabled on Modules or Modifiers and implement pre- or post-checks on each transaction that the Module or Modifier they are enabled on executes. This allows Avatars to do things like limit the scope of addresses and functions that a Module or Modifier can call or ensure a certain state is never changed by a Module or Modifier. Guards should expose the `IGuard` interface. Modules, Modifiers, and Avatars that wish to be guardable MUST inherit `Guardable`, MUST call `checkTransaction()` before triggering execution on their target, and MUST call `checkAfterExecution()` after execution is complete.
+
+```solidity
+/// @title Zodiac Avatar - A contract that manages modules that can execute transactions via this contract.
+
+pragma solidity >=0.7.0 <0.9.0;
+
+import "@gnosis.pm/safe-contracts/contracts/common/Enum.sol";
+
+
+interface IAvatar {
+    /// @dev Enables a module on the avatar.
+    /// @notice Can only be called by the avatar.
+    /// @notice Modules should be stored as a linked list.
+    /// @notice Must emit EnabledModule(address module) if successful.
+    /// @param module Module to be enabled.
+    function enableModule(address module) external;
+
+    /// @dev Disables a module on the avatar.
+    /// @notice Can only be called by the avatar.
+    /// @notice Must emit DisabledModule(address module) if successful.
+    /// @param prevModule Address that pointed to the module to be removed in the linked list
+    /// @param module Module to be removed.
+    function disableModule(address prevModule, address module) external;
+
+    /// @dev Allows a Module to execute a transaction.
+    /// @notice Can only be called by an enabled module.
+    /// @notice Must emit ExecutionFromModuleSuccess(address module) if successful.
+    /// @notice Must emit ExecutionFromModuleFailure(address module) if unsuccessful.
+    /// @param to Destination address of module transaction.
+    /// @param value Ether value of module transaction.
+    /// @param data Data payload of module transaction.
+    /// @param operation Operation type of module transaction: 0 == call, 1 == delegate call.
+    function execTransactionFromModule(
+        address to,
+        uint256 value,
+        bytes memory data,
+        Enum.Operation operation
```

Comment:
suggestion:  change `Enum.Operation` to a  basic data type such as `uint256` .

Reply:
For compatibility's sake, the interface is the same as the Gnosis Safe ModuleManager interface, so we can't really change this.